packages/payload/src/fields/config/types.ts

```diff
@@ -1104,14 +1105,21 @@ export type SelectField = {
    */
   enumName?: DBIdentifierName
   hasMany?: boolean
-  /** Customize generated GraphQL and Typescript schema names.
+  /**
+   * Customize generated GraphQL and Typescript schema names.
    * By default, it is bound to the collection.
    *
    * This is useful if you would like to generate a top level type to share amongst collections/fields.
    * **Note**: Top level types can collide, ensure they are unique amongst collections, arrays, groups, blocks, tabs.
    */
   interfaceName?: string
   options: Option[]
+  /**
+   * Reduce the available options based on the current user or value of another field.
+   * Similar to the `filterOptions` property on `relationship` and `upload` fields, and the `condition` property on all fields.
+   * Used to determine which options are shown to the user and what is validated on the server.
+   */
+  reduceOptions?: (args: { data: Data; options: Option[]; req: PayloadRequest }) => Option[]
```

Comment:
We will be asked to add siblingData on this for sure.

Reply:
Done in [677f0f0](https://github.com/payloadcms/payload/pull/12487/commits/677f0f0839bb1f7e9eca43cf42e7f918c76107fd).